src/joint_constraints.rs

```diff
@@ -116,12 +113,7 @@ impl JointConstraint {
 /// Returns None if `v` is unbound in all Bindings in `parts`.
 #[inline]
 fn resolve_var_len(parts: &[Bindings], v: char) -> Option<usize> {
-    for b in parts {
-        if let Some(s) = b.get(v) {
-            return Some(s.len());
-        }
-    }
-    None
+    parts.iter().map(|bindings| bindings.get(v).map(String::len)).reduce(Option::or).flatten()
```

Comment:
My understanding is that this loops through everything whereas the original cuts it short if it finds a `Some`. Maybe we can do something with `find_map`?